ports/geckolib/properties.mako.rs

```diff
@@ -184,6 +185,54 @@ def set_gecko_property(ffi_name, expr):
     }
 </%def>
 
+<%def name="clear_color_flags(color_flags_ffi_name)">
+    % if color_flags_ffi_name:
+    self.gecko.${color_flags_ffi_name} &= !(gecko_style_structs::BORDER_COLOR_SPECIAL as u8);
+    % endif
+</%def>
+
+<%def name="set_current_color_flag(color_flags_ffi_name)">
+    % if color_flags_ffi_name:
+    self.gecko.${color_flags_ffi_name} |= gecko_style_structs::BORDER_COLOR_FOREGROUND as u8;
+    % else:
+    // FIXME(heycam): To handle this, we would need to get the
```

Comment:
We definitely can hit this, for example with background-color:currentColor. The real solution is to make Gecko not compute down background-color:currentColor to a color, and to store currentColor as one of the computed values. I think Xidorn will look into that in https://bugzilla.mozilla.org/show_bug.cgi?id=760345 but I'm not sure when. So I think the comment should stay for now.